services/ui-src/src/components/modals/DeleteEntityModal.tsx

```diff
@@ -1,21 +1,45 @@
-import React from "react";
+import { useContext, useState } from "react";
 // components
 import { Text } from "@chakra-ui/react";
-import { Modal } from "components";
+import { Modal, ReportContext } from "components";
 
 // types
-import { AnyObject, EntityShape } from "types";
-import { renderHtml } from "utils";
+import { AnyObject, EntityShape, ReportStatus } from "types";
+import { renderHtml, useStore } from "utils";
 
 export const DeleteEntityModal = ({
+  entityType,
   selectedEntity,
   verbiage,
   modalDisclosure,
 }: Props) => {
-  const deleting = false;
+  const { report } = useStore();
+  const { updateReport } = useContext(ReportContext);
+  const { full_name } = useStore().user ?? {};
+  const [deleting, setDeleting] = useState<boolean>(false);
+
   const deleteProgramHandler = async () => {
-    // eslint-disable-next-line no-console
-    console.log("Entity Deleted", selectedEntity);
+    setDeleting(true);
+    const reportKeys = {
+      reportType: report?.reportType,
+      state: report?.state,
+      id: report?.id,
+    };
+    const currentEntities = report?.fieldData?.[entityType] || [];
+    const updatedEntities = currentEntities.filter(
+      (entity: EntityShape) => entity.id != selectedEntity?.id
```

Comment:
is the single `=` intentional?

Reply:
Just a straight copy from MCR. I can make it double!